API Keys content update (#3594)
temporalio/documentation: docs/develop/dotnet/temporal-client.mdx

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect to Temporal Cloud using an API Key

> To use your [API key](/cloud/api-keys) with a Temporal SDK, include the API key in the Bearer token header and specify the Namespace header. The Bearer token header indicates the authentication type, while the Namespace header routes the request.

Reviewer: ```suggestion
To use an [API key](/cloud/api-keys) with the Temporal .NET SDK, use the following example.
```

To create an initial connection:

```csharp
var myClient = TemporalClient.ConnectAsync(new(<endpoint>)
{
    Namespace = <namespace_id>.<account_id>,
    ApiKey = <APIKey>,

>     RpcMetadata = new Dictionary<string, string>()
>     {
>         ["temporal-namespace"] = "${<namespace_id>.<account_id>}",
>     },

Reviewer: ```suggestion

```

### How to connect to Temporal Cloud using an API Key {#connect-to-temporal-cloud-api-key}

To use an [API key](/cloud/api-keys) with the Temporal .NET SDK, you will need to gather some information first:

- When you create your _API Key,_ be sure to save the value.
- You can see the _namespace account id_ combination in your URL or anywhere that you manage namespaces.
  It follows the format `<namespace_id>.<account_id>`.
- The _endpoint_ will vary. The most common endpoint used is the gRPC regional endpoint, which you can find by selecting the Connect button on the namespace details page.
  It follows the format: `<region>.<cloud_provider>.api.temporal.io:7233`.
- For Namespaces with High Availability features with API key authentication enabled, use the gRPC Namespace endpoint: `<namespace>.<account>.tmprl.cloud:7233`.
  This allows automated failover without needing to switch endpoints.

Once you have the necessary info, to create an initial connection, use this code:

To create an initial connection:

```csharp
var myClient = TemporalClient.ConnectAsync(new(<endpoint>)
{
    Namespace = "<namespace_id>.<account_id>",
    ApiKey = "<APIKey>",
    Tls = new(),
});
```

> To update an API key:

Reviewer 2: See similar comment on Python on this "update" section — every one of these "update" examples is more or less just updating a dictionary key using their respective language features, which is almost too obvious to require documenting at all, and if we do want to keep it, I would clarify that it really is just updating a param you've already defined. I'm not sure if this is necessary.

Author: Agreed; this is almost unnecessary, but I'll add a little context.
